src/metatrain/utils/data/dataset.py

```diff
@@ -449,15 +451,19 @@ def __init__(self, path: Union[str, Path]):
         for file_name in self.zip_file.namelist():
             if file_name.startswith("0/") and file_name.endswith(".mts"):
                 self._field_names.append(file_name[2:-4])
-        self._sample_class = namedtuple("Sample", self._field_names)
+        self._fields_to_read = (
+            self._field_names if fields is None else ["system", *fields]
+        )
```

Comment:
should this also check that the request fields exist and give a good error message if not?

Reply:
Yes, for sure a good idea, done!